dir.props

```diff
@@ -71,6 +72,10 @@
   </PropertyGroup>
   <Import Project="$(ToolRuntimePath)/net45/roslyn/build/Microsoft.Net.Compilers.props" Condition="'$(UseRoslynCompilers)'!='false'" />
 
+  <PropertyGroup>
+    <_BuildAgainstPackages Condition="'$(KeepAllProjectReferences)' != 'true' and $([System.Text.RegularExpressions.Regex]::IsMatch('$(MSBuildProjectFullPath)', '\\tests\\', System.Text.RegularExpressions.RegexOptions.IgnoreCase))">$(BuildTestsAgainstPackages)</_BuildAgainstPackages>
```

Comment:
I'm not sure I like this property in the dir.props are you only putting it here. I assume you are only doing it so you can setup the other properties below. Is there any reason they cannot be put in your targets you have supporting this in the BuildTools?